python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -138,45 +133,78 @@ def get_targeted_partitioned_asset_keys_topological_order(self) -> Sequence[Asse
         targeted_toposorted_keys = []
         for level_keys in toposorted_keys:
             for key in sorted(level_keys):
-                if (
-                    key in self.target_subset.asset_keys
-                    and self.target_subset.asset_graph.get_partitions_def(key) is not None
-                ):
+                if key in self.target_subset.asset_keys:
                     targeted_toposorted_keys.append(key)
 
         return targeted_toposorted_keys
 
-    def get_partitions_status_counts_by_asset_key(
+    def get_status_by_asset_key(
         self,
-    ) -> Mapping[AssetKey, Mapping[BackfillPartitionsStatus, int]]:
-        """Returns a mapping from asset key to a mapping from status to count of partitions in that
-        status.
-
-        Only includes assets that are partitioned.
+    ) -> Mapping[
+        AssetKey,
+        Union[
+            Tuple[Mapping[BackfillPartitionsStatus, int], int],
+            Optional[BackfillPartitionsStatus],
+        ],
+    ]:
+        """Returns a mapping containing the status of each targeted asset key.
+
+        For unpartitioned asset keys, returns a single status, None if the asset has not
+        started executing.
+
+        For partitioned asset keys, returns a tuple containing a mapping of status counts
+        and the total number of partitions targeted for the asset key.
         """
 
-        def _get_status_counts_for_asset_key(asset_key: AssetKey):
-            materialized_subset = self.materialized_subset.get_partitions_subset(asset_key)
-            failed_partitions = set(
-                self.failed_and_downstream_subset.get_partitions_subset(
-                    asset_key
-                ).get_partition_keys()
-            )
-            requested_partitions = set(
-                self.requested_subset.get_partitions_subset(asset_key).get_partition_keys()
-            )
+        def _get_status_for_asset_key(
+            asset_key: AssetKey,
+        ) -> Union[
+            Tuple[Mapping[BackfillPartitionsStatus, int], int],
```

Comment:
Someone reading this code is likely to need to take some time to decipher the meaning of this data structure. Replacing it with a NamedTuple would make it clearer what's going on.